### ISSUE TITLE
Error converting dataframe to matrix 

### ISSUE THREAD

**Issue by gtollefson** [JuliaData/CSV.jl#216]
I receive the following error when attempting to convert a CSV.read generated dataframe into an array using Matrix().

I did not have this issue when converting a DataFrames.jl readtable() generated dataframe to matrix.

> MethodError: promote_rule(::Type{CategoricalArrays.CategoricalString{UInt32}}, ::Type{Any}) is ambiguous. Candidates:
>   promote_rule(::Type{C}, ::Type{T}) where {C<:(Union{CategoricalArrays.CategoricalString{R}, CategoricalArrays.CategoricalValue{T,R} where T} where R), T} in CategoricalArrays at /Users/George/.julia/v0.6/CategoricalArrays/src/value.jl:53
>   promote_rule(::Type{T}, ::Type{Any}) where T in Missings at /Users/George/.julia/v0.6/Missings/src/Missings.jl:52
> Possible fix, define
>   promote_rule(::Type{T<:(Union{CategoricalArrays.CategoricalString{R}, CategoricalArrays.CategoricalValue{T,R} where T} where R)}, ::Type{Any})

**Comment by nalimilan** [comment]
Can you post a reproducible example?

**Comment by gtollefson** [comment]
Sure thing. I've attached a csv to read as a DataFrame using CSV.read() with the following arguments:

`df_vcf = CSV.read("test_example_file.txt", delim="\t", datarow = 2, header = 1, types=Dict(1=>String))`

`vcf=Matrix(df_vcf)
`

[test_example_file.txt](https://github.com/JuliaData/CSV.jl/files/2091646/test_example_file.txt)

**Comment by gtollefson** [comment]
I attached a toy file instead of defining a toy matrix because I'm not sure how to set a column type to ::Type{CategoricalArrays.CategoricalString{UInt32}} which is necessary for the error to be reproducible.

**Comment by nalimilan** [comment]
Thanks, I've identified the problem. See https://github.com/JuliaData/CategoricalArrays.jl/pull/146. Until then, you can fix the problem locally by running this line:
```julia
Base.promote_rule(::Type{C}, ::Type{Any}) where {C <: CategoricalArrays.CatValue} = Any
```

**Comment by gtollefson** [comment]
@nalimilan Thank you for solving so quickly. However, I have tried to run this line within my script on Julia v.06 but receive the following error:

> ERROR: LoadError: MethodError: no method matching promote_rule(::Type{CategoricalArrays.CategoricalString{UInt32}}, ::Type{Any})
> The applicable method may be too new: running in world age 22025, while current world is 22026.
> Closest candidates are:
>   promote_rule(::Type{C<:(Union{CategoricalArrays.CategoricalString{R}, CategoricalArrays.CategoricalValue{T,R} where T} where R)}, ::Type{Any}) where C<:(Union{CategoricalArrays.CategoricalString{R}, CategoricalArrays.CategoricalValue{T,R} where T} where R) at /Users/George/projects_code/ViVa.jl/viva_main.jl:128 (method too new to be called from this world context.)
>   promote_rule(::Type{T}, ::Type{Any}) where T at /Users/George/.julia/v0.6/Missings/src/Missings.jl:52
>   promote_rule(::Type{T}, ::Type{Union{Missings.Missing, S}}) where {T, S} at /Users/George/.julia/v0.6/Missings/src/Missings.jl:51
>   ...
> Stacktrace:
>  [1] promote_type(::Type{Any}, ::Type{CategoricalArrays.CategoricalString{UInt32}}) at ./promotion.jl:161
>  [2] macro expansion at ./reduce.jl:188 [inlined]
>  [3] macro expansion at ./simdloop.jl:73 [inlined]
>  [4] mapreduce_impl(::Base.#identity, ::Base.#promote_type, ::Array{Type,1}, ::Int64, ::Int64, ::Int64) at ./reduce.jl:186
>  [5] _mapreduce(::Base.#identity, ::Base.#promote_type, ::IndexLinear, ::Array{Type,1}) at ./reduce.jl:280
>  [6] convert at /Users/George/.julia/v0.6/DataFrames/src/abstractdataframe/abstractdataframe.jl:615 [inlined]
>  [7] Array{T,2} where T(::DataFrames.DataFrame) at ./sysimg.jl:24
>  [8] main() at /Users/George/projects_code/ViVa.jl/viva_main.jl:129
>  [9] include_from_node1(::String) at ./loading.jl:569
>  [10] include(::String) at ./sysimg.jl:14
>  [11] process_options(::Base.JLOptions) at ./client.jl:305
>  [12] _start() at ./client.jl:371
> while loading /Users/George/projects_code/ViVa.jl/viva_main.jl, in expression starting on line 376

**Comment by nalimilan** [comment]
Looks like defining the method at the REPL doesn't work. Try `Pkg.checkout("CategoricalArrays")` (but don't forget to call `Pkg.free("CategoricalArrays")` once the fix is released to avoid staying blocked at the current version).

**Comment by gtollefson** [comment]
Ok, I've tried this but receive the error: 

`julia> Pkg.checkout("CategoricalArrays")
INFO: Checking out CategoricalArrays master...
INFO: Pulling CategoricalArrays latest master...
ERROR: fixed packages introduce conflicting requirements for Compat: 
         CSV requires versions [0.41.0,∞)
         CategoricalArrays requires versions [0.67.0,∞) [none of the available versions can satisfy this requirement]
       available versions are 0.0.1, 0.0.2, 0.1.0, 0.1.1, 0.2.0, 0.2.1, 0.2.2, 0.2.3, 0.2.4, 0.2.5, 0.2.6, 0.2.7, 0.2.8, 0.2.9, 0.2.10, 0.2.11, 0.2.12, 0.2.13, 0.3.0, 0.3.1, 0.3.2, 0.3.3, 0.3.4, 0.3.5, 0.3.6, 0.3.7, 0.3.8, 0.4.0, 0.4.1, 0.4.2, 0.4.3, 0.4.4, 0.4.5, 0.4.6, 0.4.7, 0.4.8, 0.4.9, 0.4.10, 0.4.11, 0.4.12, 0.4.13, 0.5.0, 0.5.1, 0.6.0, 0.7.0, 0.7.1, 0.7.2, 0.7.3, 0.7.4, 0.7.5, 0.7.6, 0.7.7, 0.7.8, 0.7.9, 0.7.10, 0.7.11, 0.7.12, 0.7.13, 0.7.14, 0.7.15, 0.7.16, 0.7.17, 0.7.18, 0.7.19, 0.7.20, 0.8.0, 0.8.1, 0.8.2, 0.8.3, 0.8.4, 0.8.5, 0.8.6, 0.8.7, 0.8.8, 0.9.0, 0.9.1, 0.9.2, 0.9.3, 0.9.4, 0.9.5, 0.10.0, 0.11.0, 0.12.0, 0.13.0, 0.14.0, 0.15.0, 0.16.0, 0.16.1, 0.16.2, 0.17.0, 0.18.0, 0.19.0, 0.20.0, 0.21.0, 0.22.0, 0.23.0, 0.24.0, 0.25.0, 0.25.1, 0.25.2, 0.26.0, 0.27.0, 0.28.0, 0.29.0, 0.30.0, 0.31.0, 0.32.0, 0.33.0, 0.34.0, 0.35.0, 0.36.0, 0.37.0, 0.38.0, 0.39.0, 0.40.0, 0.41.0, 0.42.0, 0.43.0, 0.44.0, 0.45.0, 0.46.0, 0.47.0, 0.48.0, 0.49.0, 0.50.0, 0.51.0, 0.52.0, 0.53.0, 0.54.0, 0.55.0, 0.55.1, 0.55.2, 0.56.0, 0.57.0, 0.58.0, 0.59.0, 0.60.0, 0.61.0, 0.62.0, 0.62.1, 0.63.0, 0.64.0, 0.65.0 and 0.65.1
Stacktrace:
 [1] check_requirements(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Base.Pkg.Types.Fixed}) at ./pkg/query.jl:222
 [2] resolve(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Base.Pkg.Types.Fixed}, ::Dict{String,VersionNumber}, ::Set{String}) at ./pkg/entry.jl:496
 [3] resolve(::Dict{String,Base.Pkg.Types.VersionSet}, ::Dict{String,Dict{VersionNumber,Base.Pkg.Types.Available}}, ::Dict{String,Tuple{VersionNumber,Bool}}, ::Dict{String,Base.Pkg.Types.Fixed}) at ./pkg/entry.jl:479
 [4] (::Base.Pkg.Entry.##16#18{String,String,Bool,Bool})(::Base.LibGit2.GitRepo) at ./pkg/entry.jl:236
 [5] transact(::Base.Pkg.Entry.##16#18{String,String,Bool,Bool}, ::Base.LibGit2.GitRepo) at ./libgit2/libgit2.jl:882
 [6] with(::Base.Pkg.Entry.##15#17{String,String,Bool,Bool}, ::Base.LibGit2.GitRepo) at ./libgit2/types.jl:608
 [7] checkout(::String, ::String, ::Bool, ::Bool) at ./pkg/entry.jl:226
 [8] (::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#checkout,Tuple{String,String,Bool,Bool}})() at ./pkg/dir.jl:36
 [9] cd(::Base.Pkg.Dir.##4#7{Array{Any,1},Base.Pkg.Entry.#checkout,Tuple{String,String,Bool,Bool}}, ::String) at ./file.jl:70
 [10] #cd#1(::Array{Any,1}, ::Function, ::Function, ::String, ::Vararg{Any,N} where N) at ./pkg/dir.jl:36
 [11] #checkout#1(::Bool, ::Bool, ::Function, ::String, ::String) at ./pkg/pkg.jl:188
 [12] checkout(::String) at ./pkg/pkg.jl:188
`

**Comment by nalimilan** [comment]
You probably need to call `Pkg.update()`.
